node/src/accountant/mod.rs

```diff
@@ -745,133 +742,103 @@ impl Accountant {
         }
     }
 
-    fn handle_report_routing_service_provided_message(
-        &mut self,
-        msg: ReportRoutingServiceProvidedMessage,
-    ) {
-        let msg_id = self.msg_id();
-        trace!(
-            self.logger,
-            "Msg {}: Charging routing of {} bytes to wallet {}",
-            msg_id,
-            msg.payload_size,
-            msg.paying_wallet
-        );
-
-        let sum = self.record_service_provided(
-            msg.service_rate,
-            msg.byte_rate,
-            msg.timestamp,
-            msg.payload_size,
-            &msg.paying_wallet,
-        );
-
-        self.logging_utils.accounting_msgs_stats.manage_debug_log(
-            &self.logger,
-            AccountingMsgType::RoutingServiceProvided,
-            DEFAULT_ACCOUNTING_MSG_LOG_WINDOW,
-            vec![NewPosting::new(msg.paying_wallet.address(), sum)],
-        )
-    }
+    fn handle_report_service_provided_message<AccountingMessage>(&mut self, msg: AccountingMessage)
+    where
+        AccountingMessage: MessageWithServicesProvided,
+    {
+        let services_provided = &msg.services_provided();
+        let accounting_msg_type = msg.msg_type();
+        let trace_log_wrapper = msg.trace_log_wrapper();
 
-    fn handle_report_exit_service_provided_message(
-        &mut self,
-        msg: ReportExitServiceProvidedMessage,
-    ) {
-        self.logger.trace(||{
-            let msg_id = self.msg_id();
-            format!(
-                "Msg {}: Charging exit service for {} bytes to wallet {} at {} per service and {} per byte",
-                msg_id,
-                msg.payload_size,
-                msg.paying_wallet,
-                msg.service_rate,
-                msg.byte_rate
-            )
-        });
-        let sum = self.record_service_provided(
-            msg.service_rate,
-            msg.byte_rate,
-            msg.timestamp,
-            msg.payload_size,
-            &msg.paying_wallet,
-        );
+        trace_log_wrapper.log_trace(self.logging_kit(MsgIdRequested::New));
```

Comment:
This code is worrisome. I think if we had to find the log once we have the log file with us, then at that time, it'll be difficult for us to find this.

This layer of abstraction is the issue, if it wasn't for a use case like logging, I'd appreciate it and here, I'm concerned.